File: backend/src/main/java/de/snickit/rezeptebackend/entities/RecipeStep.java
package de.snickit.rezeptebackend.entities;

import de.snickit.rezeptebackend.entities.pk.RecipeStepKey;

import javax.persistence.*;
import java.util.Collection;

@Entity
@Table(name = "recipe_step", schema = "public")
@IdClass(RecipeStepKey.class)
public class RecipeStep {

    @Id
    @ManyToOne
    private Recipe recipeId;

    @Id
    private Long stepId;

    @OneToMany(mappedBy = "recipeStep")
    private Collection<RecipeStepIngredients> ingredients;

    public Collection<RecipeStepIngredients> getIngredients() {
        return ingredients;
    }

    public void setIngredients(Collection<RecipeStepIngredients> ingredients) {
        this.ingredients = ingredients;
    }
}

File: backend/src/main/kotlin/de/snickit/recipe/backend/persistence/entity/RecipeStep.kt
package de.snickit.recipe.backend.persistence.entity

import javax.persistence.*

@Entity
@Table(name = "recipe_step", schema = "public", catalog = "rezepte_db")
@IdClass(RecipeStepPK::class)
class RecipeStep {
    @Column(name = "recipe_id", nullable = false, insertable = false, updatable = false)
    @Id
    var recipeId: Long = 0

    @Column(name = "step_id", nullable = false, insertable = false, updatable = false)
    @Id
    var stepId = 0

    @Column(name = "title", nullable = true, length = 255)
    @Basic
    var title: String? = null

    @Column(name = "description", nullable = true, length = 20000)
    @Basic
    var description: String? = null

    @JoinColumn(name = "recipe_id", referencedColumnName = "recipe_id", nullable = false, insertable = false, updatable = false
    )
    @ManyToOne
    var recipeByRecipeId: Recipe? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return true
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}
